x-pack/plugins/apm/public/components/shared/Links/MachineLearningLinks/MLLink.tsx

```diff
@@ -45,8 +43,14 @@ export function MLLink({ children, path = '', query = {}, external }: Props) {
       jobId: jobIds,
       groupIds: ['apm'],
       globalState: {
-        time,
-        refreshInterval,
+        time:
+          rangeFrom !== undefined && rangeTo !== undefined
+            ? { from: rangeFrom, to: rangeTo }
+            : undefined,
+        refreshInterval:
+          refreshPaused !== undefined && refreshInterval !== undefined
+            ? { pause: refreshPaused, value: refreshInterval }
+            : undefined,
```

Comment:
Since `time` and `refreshInterval` are already optional, would it be possible to make `from`, `to`, `pause` and `value` optional too? Or does that complicate it too much on your side (for some reason I can't find the implementation of `useMlHref` - it's in master already, right?)

```suggestion
        time: { from: rangeFrom, to: rangeTo },
        refreshInterval: { pause: refreshPaused, value: refreshInterval }
```

Reply:
Btw. thanks for making these changes!

Reply:
I did some testing on the ML end and making them optional requires a bit of work as it's not currently supported. Since we will be creating a synchronous version of the mlUrlGenerator anyway, I'll add this to the meta issue and we can do this in a follow-up PR. What do you think?

Reply:
Sounds good to me 👍